presto-parser/src/main/java/com/facebook/presto/sql/ExpressionFormatter.java

```diff
@@ -566,6 +575,50 @@ static String formatSortItems(List<SortItem> sortItems, boolean unmangleNames)
                 .iterator());
     }
 
+    static String formatGroupBy(List<GroupingElement> groupingElements)
+    {
+        ImmutableList.Builder<String> resultStrings = ImmutableList.builder();
+
+        for (GroupingElement groupingElement : groupingElements) {
+            String result = "";
+            if (groupingElement instanceof SimpleGroupBy) {
+                Set<Expression> columns = ImmutableSet.copyOf(((SimpleGroupBy) groupingElement).getColumnExpressions());
+                if (columns.size() == 1) {
+                    result = formatExpression(getOnlyElement(columns));
+                }
+                else {
+                    result = format("(%s)", formatGroupingSet(columns));
+                }
+            }
+            else if (groupingElement instanceof GroupingSets) {
+                result = format("GROUPING SETS (%s)", Joiner.on(", ").join(
+                        groupingElement.enumerateGroupingSets().stream()
+                                .map(ExpressionFormatter::formatGroupingSet)
+                                .iterator()));
+            }
+            else if (groupingElement instanceof Cube) {
+                result = format("CUBE (%s)", formatGroupingSet(((Cube) groupingElement).getColumns()));
```

Comment:
This will produce `CUBE ((a, b, c))`. Don't we want `CUBE (a, b, c)` in this case?